packages/amazonq/test/unit/amazonq/apps/inline/completion.test.ts

```diff
@@ -400,57 +411,71 @@ describe('InlineCompletionManager', () => {
 
                     assert.strictEqual(result[0].insertText, expectedText)
                 }),
-                describe('debounce behavior', function () {
-                    let clock: ReturnType<typeof installFakeClock>
-
-                    beforeEach(function () {
-                        clock = installFakeClock()
+                it('shows message to user when manual invoke fails to produce results', async function () {
+                    provider = new AmazonQInlineCompletionItemProvider(
+                        languageClient,
+                        recommendationService,
+                        mockSessionManager,
+                        inlineTutorialAnnotation,
+                        true
+                    )
+                    getActiveRecommendationStub.returns([])
+                    let messageShown = false
+                    getTestWindow().onDidShowMessage((e) => {
+                        assert.strictEqual(e.message, noInlineSuggestionsMsg)
+                        messageShown = true
                     })
+                    await provider.provideInlineCompletionItems(
+                        mockDocument,
+                        mockPosition,
+                        { triggerKind: InlineCompletionTriggerKind.Invoke, selectedCompletionInfo: undefined },
+                        mockToken
+                    )
+                    // Wait up to a second for message to appear since there is potential race condition.
+                    await waitUntil(async () => messageShown, { timeout: 1000, interval: 100 })
```

Comment:
instead of waiting I think you might be able to do something like this:
```
const p = new Promise((resolve) => {
     getTestWindow().onDidShowMessage((e) => {
         assert.strictEqual(e.message, noInlineSuggestionsMsg)
         resolve()
     })
})
```
and then
```
await p
```
here